### PR TITLE
Fix Mass+ Search By Tags

### LINKED ISSUE #327: Mass+ won't search by tags anymore.
It's not possible to search for a tag on your posts in order to delete or edit them.

Chrome, newest version.

### PR DESCRIPTION
**Pull request by blackjackkent**
Fixes #327

### CODE REVIEW

File: Extensions/mass_plus.js [hunk 1]
```diff
@@ -227,60 +166,42 @@ XKit.extensions.mass_plus = new Object({
...
-							if (mr_tag.toLowerCase() === tag || r_tags[i].toLowerCase() === tag.toLowerCase()) {
-								var m_id = $(this).attr('data-post-id');
-								XKit.extensions.mass_plus.search_found_posts.push(m_id);
-								XKit.extensions.mass_plus.search_found_count++;
-							}
-						}
-					}
+				var response_object = JSON.parse(response.response);
+				var posts_array = response_object.response.posts;
+				if (!posts_array || posts_array.length == 0) {
```
Comment (ThePsionic) [inline Extensions/mass_plus.js]: This `==` should be a `===`.

**Comment by blackjackkent** [comment]
Sorry, Travis. :P Updated.

File: Extensions/mass_plus.js [hunk 2]
```diff
@@ -227,60 +166,42 @@ XKit.extensions.mass_plus = new Object({
...
-		if (m_search_url.indexOf("/") !== -1) { m_search_url = m_search_url.substring(0, m_search_url.indexOf("/")); }
-		if (m_search_url.indexOf("?") !== -1) { m_search_url = m_search_url.substring(0, m_search_url.indexOf("?")); }
-		if (m_search_url.indexOf("#") !== -1) { m_search_url = m_search_url.substring(0, m_search_url.indexOf("#")); }
-
-		this.search_url = "http://" + m_search_url + ".tumblr.com/archive";
-
+		var blog_shortname = document.location.href.substring(document.location.href.indexOf('mega-editor/') + 12);
+		if (blog_shortname .indexOf("/") !== -1) { blog_shortname = blog_shortname.substring(0, blog_shortname.indexOf("/")); }
+		if (blog_shortname .indexOf("?") !== -1) { blog_shortname = blog_shortname.substring(0, blog_shortname.indexOf("?")); }
+		if (blog_shortname .indexOf("#") !== -1) { blog_shortname = blog_shortname.substring(0, blog_shortname.indexOf("#")); }
```
Comment (bvtsang) [inline Extensions/mass_plus.js]: Nitpicks: spaces between `blog_shortname` and `.indexOf`

File: Extensions/mass_plus.js [hunk 3]
```diff
@@ -227,60 +166,42 @@ XKit.extensions.mass_plus = new Object({
...
+					return;
+				}
+				posts_array.forEach(function(post) {
+					XKit.extensions.mass_plus.search_found_posts.push(post.id);
+					XKit.extensions.mass_plus.search_found_count++;
 				});
 				if (XKit.extensions.mass_plus.search_found_count < 100 && XKit.extensions.mass_plus.search_page <= 5) {
-					var next_link = $("#next_page_link", response.responseText).attr('href');
-					if (typeof next_link === "undefined") {
+					if (posts_array.length < 20) {
```
Comment (bvtsang) [inline Extensions/mass_plus.js]: What is the significant of the number 20? Is it the default number of posts to be returned from the [`/posts`](https://www.tumblr.com/docs/en/api/v2#posts) API call? Regardless, we should be clear about what this number (and the number on line 181, assuming they're for the same purpose) represents.

**Comment by bvtsang** [comment]
LGTM

@homu r+

**Comment by 0xazure** [comment]
@homu retry

**Pull request merged**